packages/compass-collection/src/components/collection-header/collection-header.tsx

```diff
@@ -1,33 +1,77 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import type AppRegistry from 'hadron-app-registry';
+import { css } from '@mongodb-js/compass-components';
+import type { Document } from 'mongodb';
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import toNS from 'mongodb-ns';
 
 import CollectionHeaderActions from '../collection-header-actions';
 import ReadOnlyBadge from './read-only-badge';
 import TimeSeriesBadge from './time-series-badge';
 import ViewBadge from './view-badge';
 
-import styles from './collection-header.module.less';
+const collectionHeaderStyles = css({
+  paddingTop: '15px',
+  paddingBottom: '5px',
+});
 
-class CollectionHeader extends Component {
-  static displayName = 'CollectionHeaderComponent';
+const collectionHeaderNamespaceStyles = css({
+  display: 'contents',
+});
 
-  static propTypes = {
-    globalAppRegistry: PropTypes.object.isRequired,
-    namespace: PropTypes.string.isRequired,
-    isReadonly: PropTypes.bool.isRequired,
-    isTimeSeries: PropTypes.bool.isRequired,
-    statsPlugin: PropTypes.func.isRequired,
-    selectOrCreateTab: PropTypes.func.isRequired,
-    statsStore: PropTypes.object.isRequired,
-    sourceName: PropTypes.string,
-    sourceReadonly: PropTypes.bool.isRequired,
-    sourceViewOn: PropTypes.string,
-    editViewName: PropTypes.string,
-    pipeline: PropTypes.array
-  };
+const collectionHeaderTitleStyles = css({
+  fontSize: '24px',
+  fontWeight: 'normal',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  display: 'flex',
+  paddingLeft: '15px',
+  paddingRight: '15px',
+  margin: 0,
+  lineHeight: '32px',
+  alignItems: 'center',
+});
+
+const collectionHeaderTitleDBStyles = css({
+  color: ' #337ab7',
+  flexShrink: 2,
+  flexBasis: 'auto',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+  cursor: 'pointer',
+  textDecoration: 'none',
+  '&:hover,&:focus': {
+    textDecoration: 'underline',
+  },
+});
+
+const collectionHeaderTitleCollectionStyles = css({
+  flexBasis: 'auto',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+});
+
+type CollectionHeaderProps = {
+  globalAppRegistry: AppRegistry;
+  namespace: string;
+  isReadonly: boolean;
+  isTimeSeries: boolean;
+  statsPlugin: React.FunctionComponent<any>;
+  selectOrCreateTab: React.FunctionComponent<any>;
```

Comment:
This doesn’t look as if it’s used as a component, but as an action (I think that’s the right term – never quite sure about these things … :slightly_smiling_face:)

Either way, I think typing it as `React.FunctionComponent` doesn’t quite communicate the right thing, even if the type might technically fit. I get that it’s a lot of work to actually go and replace all the `any` with concrete types, but I think this should be fine?

```suggestion
  selectOrCreateTab: (options: any) => any;
```

Reply:
For some `any` i really put an effort to figure out what should be used instead. In some cases, I didn't fully understand from where a prop is coming, therefore instead of confusing people later, who will take a look at the types I introduce here :) I left them as `any` with the hope that later someone who will work with this package and knows for sure what it is replaces those any.

With this particular case yeah I agree! Will push this change with the next commit.